src/cdomains/arrayDomain.ml

```diff
@@ -563,6 +563,27 @@ struct
 
   let update_length _ x = x
 end
+(* This is the main array out of bounds check*)
+let array_oob_check ( type a ) (module Idx: IntDomain.Z with type t = a) (x, l) (e, v) =
+  if GobConfig.get_bool "ana.arrayoob" then (* The purpose of the following 2 lines is to give the user extra info about the array oob *)
+    let idx_before_end = Idx.to_bool (Idx.lt v l) (* check whether index is before the end of the array *)
+    and idx_after_start = Idx.to_bool (Idx.ge v (Idx.of_int Cil.ILong BI.zero)) in (* check whether the index is non-negative *)
```

Comment:
Maybe use `Cilfacade.ptrdiffikind ()` here instead of `Cil.ILong`? `Idx` uses that in the definition of its default `ikind`.